scripts/ci/build.sh

```diff
@@ -7,6 +7,15 @@ source ./scripts/ci/common.sh
 # URL to the Pulumi conversion service.
 export PULUMI_CONVERT_URL="${PULUMI_CONVERT_URL:-$(pulumi stack output --stack pulumi/tf2pulumi-service/production url)}"
 
+echo "Generating API docs for all packages for PR preview..."
+echo ""
+go install github.com/pulumi/docs/tools/resourcedocsgen@master
+resourcedocsgen docs registry \
```

Comment:
Yeah if we could limit this to just a couple of packages, like aws and aiven or something, that'd be ideal, from an iteration-speed perspective.

Reply:
So the full build took 22 mins. with all the providers. I am slightly inclined to leave it that way. I thought it was going to be more than 30 mins for the PR build based on what we see in the `docs` repo. Maybe it's just me but I don't feel 22 mins is _that_ bad?

Reply:
Let me know what you think. I am hoping we can try it out with a full docs build to favor having all docs in the preview. Of course, if it becomes a major pain point for PRs we can certainly change the command to generate docs for just a handful of them.

Reply:
To me, it all comes down to who'll be consuming these PRs. Compared to 2-3 minutes for a PR preview today, having to wait 20 minutes per commit is a pretty big deal, especially when you're iterating on a PR and want to be able to view and share preview links with others. Speed of iteration is important here, IMO, so I'd fairly strongly prefer we continued to optimize for that, and made it _possible_ for people to run full builds if they want to. After all, it's highly unlikely anyone's going to be looking closely at long-tail API-docs content in any PR here; more likely, they're going to spot a few things in a provider or two, and move on. So I'd really like to keep the speed of iterating if at all possible, since Docs PRs will always be complete. What do you think?

Reply:
> It also seems unlikely folks will be looking closely at long-tail API-docs content in any one PR

That's a good point. How about `aws`, `azure-native`, `gcp` and `kubernetes`? They are also the featured packages right now. I can see how long those take. Should be less than a minute.

Reply:
Awesome, sounds good. An extra minute or so seems like a totally reasonable thing in exchange for the value of having a few representative packages, for sure.

Reply:
> An extra minute or so seems like a totally reasonable thing in exchange for the value of having a representative package or two.

So it turns out that just generating, building, _and_ copying the docs for those packages takes 10 mins. With the Hugo build being the bottleneck. When I slim down the list to just two big providers `aws`, and `azure-native`, the total time comes down to 8 mins. With just `aiven` and `aws` it is 5 mins. I am inclined to leave it at `aws` and `aiven` for now and will leave expanding that list for another day if we feel that we want to do that.

By the way, in all of those trials, _generating_ the docs using the `resourcedocsgen` tool was very quick in contrast with the other things involved in building the site. It probably comes as no surprise that the more pages we add to Hugo, the longer it takes to build.

Reply:
That sounds good to me. Thanks @praneetloke!